### PR TITLE
:wrench: add dprint to builtins formatter

### PR DESCRIPTION
**Pull request by hemedani**

### CODE REVIEW

**Comment by hemedani** [comment]
what can I do to pass Check codestyle ?

File: lua/null-ls/builtins/formatting/dprint.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		"toml",
+		"rust",
+		"roslyn",
+	},
+	generator_opts = {
+		command = "dprint",
+		args = {
+			"fmt",
+			"--config",
+			"https://raw.githubusercontent.com/hemedani/dotfiles/main/dprint/dprint.json",
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/formatting/dprint.lua:29]: Are you sure that we want to include this into the built-in's definition? It seems like this should be left up to the user.

**Review by jose-elias-alvarez — COMMENTED** [review]
Thanks for the PR! I had one clarifying question.

> what can I do to pass Check codestyle ?

Did you format the files with `stylua` 14.2 using this project's config? When I ran your file through it locally, it did something to the whitespace.

File: lua/null-ls/builtins/formatting/dprint.lua (hunk 1 above)
Comment (hemedani) [reply lua/null-ls/builtins/formatting/dprint.lua:29]: I'm using `stylua` 0.10.1,
yes it is better to leave 
```
"
            --config",
            "https://raw.githubusercontent.com/hemedani/dotfiles/main/dprint/dprint.json",
```
to the user.
but users should initialize dprint in every project with `dprint init`.

Comment (hemedani) [reply lua/null-ls/builtins/formatting/dprint.lua:29]: I'm fixing these and pushing it again

Comment (jose-elias-alvarez) [reply lua/null-ls/builtins/formatting/dprint.lua:29]: Sure, we can mention that users should run `dprint init` in the documentation.

**Comment by jose-elias-alvarez** [comment]
Looks good! Based on what you mentioned, I think we should add a mention of the need to run `dprint init` in a project before running this, but let me know what you think.

**Comment by hemedani** [comment]
I'm adding a note for that

**Comment by jose-elias-alvarez** [comment]
Great, thank you!

**Pull request merged**